src/EditorFeatures/Core/EditAndContinue/ActiveStatementTrackingService.cs

```diff
@@ -133,16 +135,16 @@ public void EndTracking()
             _cancellationSource.Cancel();
             _cancellationSource.Dispose();
 
-            _workspace.DocumentOpened -= DocumentOpened;
-            _workspace.DocumentClosed -= DocumentClosed;
+            _documentOpenedHandlerDisposer.Dispose();
+            _documentClosedHandlerDisposer.Dispose();
```

Comment:
aside: are these disposers double dispose safe?

Reply:
I'll go ahead and make a change to WorkspaceEventRegistration to make it explicitly safe to be called multiple times.